Sources/MusicXML/Complex Types/MIDIDevice.swift

```diff
@@ -9,15 +9,39 @@
 /// DeviceName meta event, there can be multiple midi-device elements per MusicXML part starting in
 /// MusicXML 3.0.
 public struct MIDIDevice {
-    public let value: String
+    public var value: String?
     /// The optional port attribute is a number from 1 to 16 that can be used with the unofficial
     /// MIDI port (or cable) meta event.
-    public let port: Int?
+    public var port: Int?
     /// The optional id attribute refers to the score-instrument assigned to this device. If
     /// missing, the device assignment affects all score-instrument elements in the score-part.
-    public let id: String?
+    public var id: String?
 }
 
 extension MIDIDevice: Equatable { }
-extension MIDIDevice: Codable { }
+extension MIDIDevice: Codable {
+    enum CodingKeys: String, CodingKey {
+        case value = ""
+        case port
+        case id
+    }
+
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        try container.encode(value, forKey: .value)
+        if let port = port {
+            try container.encode(String(port), forKey: .port)
+        }
+        try container.encodeIfPresent(id, forKey: .id)
+    }
+
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.value = try container.decodeIfPresent(String.self, forKey: .value)
+        if let portString = try container.decodeIfPresent(String.self, forKey: .port) {
```

Comment:
Can we look for an `Int` directly here, à la:

```Swift
self.port = try container.decodeIfPresent(Int.self, forKey: .port)
```

Reply:
From the definition and confirmed by xml generated by MuseScore, the `port` is encoded as a string, e.g. `CDATA`. We should respect it to be a string.
https://github.com/w3c/musicxml/blob/gh-pages/schema/common.mod#L1023

Reply:
Ah, good call. Would it make sense to model it as a `String`, then?

Reply:
Maybe but since the comment says the port is contained to be an Int despite its raw value is a string, I feel it is better to represent it by the true logical value type. But I'll leave that to your discretion to avoid violating the design principle you have. Feel free to edit it to string.

> The optional port attribute is a number from 1 to 16 that can be used with the unofficial MIDI port (or cable) meta event.

Reply:
What is a bit strange is that, in the docs, the `port` is a [`midi-16`](https://usermanuals.musicxml.com/MusicXML/Content/ST-MusicXML-midi-16.htm) value (AKA positive integers from 1 to 16).

https://usermanuals.musicxml.com/MusicXML/Content/CT-MusicXML-midi-device.htm

Any idea why it would be `CDATA`, when, e.g., [`midi-channel`](https://github.com/w3c/musicxml/blame/gh-pages/schema/common.mod#L1043), which is also a `midi-16` value, is encoded as `PCDATA`?

Reply:
Ah, I see the MuseScore output from the `ScorePartTests`. That seems strange to me, but you are right on all counts.

Reply:
Great question! I think all attributes have to be quoted.
Just seeing some examples around https://www.w3schools.com/xml/xml_dtd_el_vs_attr.asp

Reply:
Ha! So it's less solved than I thought. My intuition is that this should just be an `Int` (gosh darn it!). I'm surely happy with this though. It's safe and respects everyone's wishes, even if their wishes are misguided.

Reply:
Actually I was wrong. Attributes are quoted but XMLCoder already takes care of it. So we can keep this as `Int`. I have a follow up #88 PR to delete these unnecessary coder and decoder.